qa/rpc-tests/nov152018_forkactivation.py

```diff
@@ -0,0 +1,277 @@
+                    tx.vout[2] = CTxOut(0, CScript([padding, OP_RETURN]))
+                    push_size += 1
+
+                assert_equal(len(tx.serialize()), tx_size)
+
+            # Make it the same format as transaction added for padding and save the size.
+            # It's missing the padding output, so we add a constant to account for it.
+            tx.rehash()
+
+            # Add the transaction to the block
+            self.add_transactions_to_block(block, [tx])
+
+            # Now that we added a bunch of transaction, we need to recompute
+            # the merkle root.
+            block.hashMerkleRoot = block.calc_merkle_root()
+
+        # Do PoW, which is cheap on regnet
+        block.solve()
+        self.tip = block
+        self.block_heights[block.sha256] = height
+        assert number not in self.blocks
+        self.blocks[number] = block
+        return block
+
+    def get_tests(self):
+        self.set_test_params()
+
+        node = self.nodes[0]
+        self.genesis_hash = int(node.getbestblockhash(), 16)
+        self.block_heights[self.genesis_hash] = 0
+        spendable_outputs = []
+
+        # save the current tip so it can be spent by a later block
+        def save_spendable_output():
+            spendable_outputs.append(self.tip)
+
+        # get an output that we previously marked as spendable
+        def get_spendable_output():
+            return PreviousSpendableOutput(spendable_outputs.pop(0).vtx[0], 0)
+
+        # returns a test case that asserts that the current tip was accepted
+        def accepted(newtip=None):
+            if newtip == None:
+                return TestInstance([[self.tip, True]])
+            else:
+                return TestInstance([[newtip, True]])
+
+        # returns a test case that asserts that the current tip was rejected
+        def rejected(reject=None):
+            if reject is None:
+                return TestInstance([[self.tip, False]])
+            else:
+                return TestInstance([[self.tip, reject]])
+
+        # move the tip back to a previous block
+        def tip(number):
+            self.tip = self.blocks[number]
+
+        # adds transactions to the block and updates state
+        def update_block(block_number, new_transactions=[]):
+            block = self.blocks[block_number]
+            self.add_transactions_to_block(block, new_transactions)
+            old_sha256 = block.sha256
+            block.hashMerkleRoot = block.calc_merkle_root()
+            block.solve()
+            # Update the internal state just like in next_block
+            self.tip = block
+            if block.sha256 != old_sha256:
+                self.block_heights[block.sha256] = self.block_heights[old_sha256]
+                del self.block_heights[old_sha256]
+            self.blocks[block_number] = block
+            return block
+
+        # shorthand for functions
+        block = self.next_block
```

Comment:
I find this pretty confusing. `block` is used both as a `CBlock` as well as the method `next_block()`. Wouldn't it be better to make this explicit?